Refactor BannedWord Service

hissy opened the pull request:
Fixes #8202 for version 9.

* Switched to Doctrine Entity
* Remove unused and old-school functions marked as deprecated on #9070
* Support wildcard `*` to make it useful to add Japanese Word. CJK languages don't have white spaces. Refs #9301
* Added a test case

hissy commented:
I can update the view also. Any suggestions?

![Screen Shot 2021-01-03 at 22 47 47](https://user-images.githubusercontent.com/514294/103480127-b9000c80-4e15-11eb-8eb5-36821a361fbe.png)

aembler commented:
Ugh. Someone was working on this for v9, but apparently they never got around to actually finishing it. 🤦

aembler commented:
I'd love to have some help on this... I think whatever is easiest, honestly.  

It's kind of weird that @6tematik reported that they couldn't reproduce this error: https://github.com/concrete5/concrete5/pull/8626

hissy commented:
Hmm. This page should be displayed like this, right?
https://github.com/concrete5/concrete5/pull/8626#issuecomment-627365279

I'll try to find the reason.

hissy commented:
Sorry, I'll check it later.

hissy commented:
@aembler 

I've fixed the broken interface. The reason for this issue is this change https://github.com/concrete5/concrete5/commit/d8afb5e6e9eaa9929cddf9077a309d627c193919 so I moved `div.ccm-dashboard-header-buttons` to outside of the form.

Now, this pull request becomes ready to merge, but should I update the name of the database migration class? It might be too old to merge.

hissy commented:
Oops, @deek87 already fixed the interface on #9531 . You should choose his one instead of mine when you merge this PR.

hissy commented:
@aembler Should I re-create this pull request against to 9.0rc branch?

aembler commented:
If there are aspects of this pull request that were not in #9531 we should probably fix them in a new pull request.
